### DISCUSSION TITLE
Surprising behaviour of SyntaxNode Update-Overloads

### DISCUSSION THREAD

**Discussion by bernd5** [dotnet/roslyn#51986] [Q&A] [answered]
I usually try to use the `Update` functions after multiple updates in CSharpSyntaxRewriters to avoid unnecessary temporary nodes instead of WithXXX-calls. 

But:
If you look for example in `LocalDeclarationStatementSyntax` which has this Update overload:

```CSharp
        public LocalDeclarationStatementSyntax Update(SyntaxTokenList modifiers, VariableDeclarationSyntax declaration, SyntaxToken semicolonToken)
            => Update(awaitKeyword: default, usingKeyword: default, modifiers, declaration, semicolonToken);
```

The usage of `default` is very surprising to me because it removes silently simply all non explicit mentioned properties of the node.

Shouldn't it be:
```CSharp
        public LocalDeclarationStatementSyntax Update(SyntaxTokenList modifiers, VariableDeclarationSyntax declaration, SyntaxToken semicolonToken)
            => Update(awaitKeyword: this.AwaitKeyword, usingKeyword: this.UsingKeyword, modifiers, declaration, semicolonToken);
```

Or is this behaviour intended? If I wanted those nodes/tokens to be removed I could use the SyntaxFactory directly.

### Update:
Those overloads seem to be introduced by this [commit](https://github.com/dotnet/roslyn/commit/5aa701241d93a78567ec0516d3f7cea1428a605d#diff-11db9eb774f896e133f70c214c5f74cd6d448fcbb286e7345c734cab2f20e204). I couldn't find any reference to them...

I guess it was not really intended - just copied from the SyntaxFactory overloads where the usage of `default` is perfectly fine. In my eyes it should be written like in `NamespaceDeclarationSyntax`.

@CyrusNajmabadi: can we change those?

**Comment by CyrusNajmabadi** [comment]
Yes you can change these.

**Comment by RikkiGibson** [comment, marked as answer]
I think 'default' should not be used as an argument in `SyntaxFactory.Update*` methods. Could you please file an issue in Roslyn to track fixing this?

**Reply by bernd5** [reply]
I created a small issue for it: https://github.com/dotnet/roslyn/issues/51997

**Reply by bernd5** [reply]
I created a PR for it...
